documentation-site/components/header-navigation.js

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import * as React from 'react';
+import Link from 'next/link';
+import {Block} from 'baseui/block';
+import {Tag} from 'baseui/tag';
+import {
+  HeaderNavigation,
+  StyledNavigationList as NavigationList,
+  ALIGN,
+} from 'baseui/header-navigation';
+import {styled} from 'baseui';
+import Menu from 'baseui/icon/menu';
+import Logo from '../images/base-web.svg';
+import GithubLogo from './github-logo';
+import SlackLogo from './slack-logo';
+import Search from './search';
+import {version} from '../../package.json';
+
+export const HEADER_BREAKPOINT = '@media screen and (min-width: 640px)';
```

Comment:
Feels kind of bad to specify this outside of the theme - Does base ui not provide enough here or was this just the most straightforward approach?

Reply:
The closest breakpoint was `medium: 600px` and that's too narrow for my use-case. I could override the theme and add more breakpoints but I don't see a problem to add arbitrary number of breakpoints as long as they all use mobile-first min-width approach. 

One downside is the array-like API that we have in `Block` - it makes sense to have a small set of breakpoints there but if those don't cut it for you, it should be safe to use custom?

Reply:
Or if we want to enforce a limited set of breakpoints we would probably need like 6-8 of them. Three breakpoints are probably not enough to fit all UIs.

Reply:
Yeah i understand. I wouldn't want to be burdened by the need to specify all values in the block array. This is fine with me, could be something we can point to for how to approach responsive